sync-global-pullsecret/sync-global-pullsecret.go

```diff
@@ -0,0 +1,263 @@
+package syncglobalpullsecret
+
+// sync-global-pullsecret syncs the pull secret from the user provided pull secret in DataPlane and appends it to the HostedCluster PullSecret to be deployed in the nodes of the HostedCluster.
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	hyperapi "github.com/openshift/hypershift/support/api"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/coreos/go-systemd/dbus"
+	"github.com/spf13/cobra"
+)
+
+// syncGlobalPullSecretOptions contains the configuration options for the sync-global-pullsecret command
+type syncGlobalPullSecretOptions struct {
+	kubeletConfigJsonPath string
+	globalPSSecretName    string
+	checkInterval         time.Duration
```

Comment:
is checkInterval used at all?

Reply:
It was, not needed anymore. Dropped already. Thanks!